addons/crm/models/crm_lead.py

```diff
@@ -468,16 +468,26 @@ def _fields_view_get(self, view_id=None, view_type='form', toolbar=False, submen
     # Actions Methods
     # ----------------------------------------
 
-    def action_set_lost(self):
+    def action_set_lost(self, **additional_values):
         """ Lost semantic: probability = 0 or active = False """
-        result = self.write({'active': False, 'probability': 0})
+        additional_values = additional_values if len(additional_values) > 0 else {}
+        result = self.write({'active': False, 'probability': 0, **additional_values})
         return result
 
     def action_set_won(self):
         """ Won semantic: probability = 100 (active untouched) """
         for lead in self:
             stage_id = lead._stage_find(domain=[('is_won', '=', True)])
-            lead.write({'stage_id': stage_id.id, 'probability': 100})
+            lead.write({'stage_id': stage_id.id, 'probability': 100, 'lost_reason': ""})
+        return True
+    
+    def action_restore(self):
+        """ Restore a lead/Opportunity by setting active to True and removing the Lost_reason """
+        for lead in self:
+            if lead.stage_id.id == 4:
```

Comment:
Can we have something dynamic instead of  comparison with `== 4 `? (Maybe xml_id if it has)

Reply:
I changed it to if lead.stage_id.is_won == True

Reply:
Or just if.lead.stage_id.is_won:

Reply:
indeed